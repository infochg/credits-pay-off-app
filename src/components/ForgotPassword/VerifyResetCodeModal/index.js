import React from 'react';
import { Form, Field, reduxForm } from 'redux-form';
import {
  Button,
  CircularProgress,
  Dialog,
  Grid,
  Typography
} from '@material-ui/core';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '../../Common/TextField';
import ErrorBoundary from '../../../containers/ErrorBoundary';

const useStyles = makeStyles(theme => ({
  modalWrapper: {
    zIndex: '9001 !important'
  },
  backDrop: {
    backgroundColor: 'rgba(255, 255, 255, 0.8)'
  },
  modal: {
    padding: '40px',
    maxWidth: '560px',
    textAlign: 'center'
  },
  modalTitle: {
    fontSize: '22px',
    color: theme.palette.text.primary,
    fontWeight: '900'
  },
  modalSubTitle: {
    fontSize: '18px',
    color: theme.palette.text.secondary
  },
  label: {
    textAlign: 'left'
  },
  modalBtns: {
    paddingTop: '20px',
    width: '480px'
  },
  modalBtn: {
    width: '144px',
    padding: '6px 10px',
    margin: '10px 20px'
  },
  [theme.breakpoints.down('xs')]: {
    form: {
      position: 'relative',
      paddingBottom: '50px'
    }
  }
}));

const validate = values => {
  const errors = {};
  if (!values.code || values.code === '') {
    errors.code = 'Field is required.';
  }
  return errors;
};

function VerifyResetCodeModal(props) {
  const classes = useStyles();
  const { handleSubmit, pristine, submitting, loading } = props;

  try {
    return (
      <Dialog
        className={classes.modalWrapper}
        open
        BackdropProps={{
          classes: {
            root: classes.backDrop
          }
        }}
      >
        <div className={classes.modal}>
          <Typography className={classes.modalTitle}>
            Please enter Code.
          </Typography>
          <Typography className={classes.modalSubTitle}>
            We have sent you SMS with the code.
          </Typography>
          <Form onSubmit={handleSubmit}>
            <Grid container alignItems="center">
              <Grid item xs={12} sm={5}>
                <Typography className={classes.label}>
                  Six-digit code:
                </Typography>
              </Grid>
              <Grid item xs={12} sm={7}>
                <Field
                  name="code"
                  component={TextField}
                  variant="outlined"
                  type="text"
                  fullWidth
                />
              </Grid>
            </Grid>
            {!loading ? (
              <Button
                variant="contained"
                color="primary"
                disableElevation
                type="submit"
                disabled={pristine || submitting}
                className={classes.submitBtn}
              >
                Send Code
              </Button>
            ) : (
              <div className={classes.submitBtn}>
                <CircularProgress />
              </div>
            )}
          </Form>
        </div>
      </Dialog>
    );
  } catch (e) {
    return <ErrorBoundary error={e} />;
  }
}

VerifyResetCodeModal.propTypes = {
  handleSubmit: PropTypes.func,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool,
  loading: PropTypes.bool
};

VerifyResetCodeModal.defaultProps = {
  handleSubmit: undefined,
  pristine: undefined,
  submitting: undefined,
  loading: undefined
};

export default reduxForm({
  form: 'VerifyResetCodeModal',
  validate
})(VerifyResetCodeModal);
