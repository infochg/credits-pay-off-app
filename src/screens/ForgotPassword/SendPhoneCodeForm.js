import React from 'react';
import { Field, reduxForm, Form } from 'redux-form';
import { Button, CircularProgress, Grid } from '@material-ui/core';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '../../components/Common/TextField';
import ErrorBoundary from '../../containers/ErrorBoundary';

const required = value => (value && value !== '' ? undefined : 'Required');

const useStyles = makeStyles(theme => ({
  submitBtn: {
    marginTop: '40px'
  },
  forgotText: {
    textAlign: 'center',
    '& a': {
      textDecoration: 'none',
      color: theme.palette.primary.main,
      '&:hover': {
        textDecoration: 'underline'
      }
    }
  }
}));

function SendPhoneCodeForm(props) {
  const classes = useStyles();
  const { handleSubmit, pristine, submitting, loading } = props;

  try {
    return (
      <Form onSubmit={handleSubmit}>
        <Grid container>
          <Grid item xs={12}>
            <Field
              name="code"
              component={TextField}
              placeholder="Email or Phone number"
              validate={required}
              variant="outlined"
              fullWidth
            />
          </Grid>
        </Grid>
        {!loading ? (
          <Button
            variant="contained"
            color="primary"
            disableElevation
            type="submit"
            disabled={pristine || submitting}
            className={classes.submitBtn}
          >
            Send Code
          </Button>
        ) : (
          <div className={classes.submitBtn}>
            <CircularProgress />
          </div>
        )}
      </Form>
    );
  } catch (e) {
    return <ErrorBoundary error={e} />;
  }
}

SendPhoneCodeForm.propTypes = {
  handleSubmit: PropTypes.func,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool,
  classes: PropTypes.shape({}),
  loading: PropTypes.bool
};

SendPhoneCodeForm.defaultProps = {
  handleSubmit: undefined,
  pristine: undefined,
  submitting: undefined,
  classes: undefined,
  loading: undefined
};

export default reduxForm({
  form: 'SendPhoneCodeForm'
})(SendPhoneCodeForm);
