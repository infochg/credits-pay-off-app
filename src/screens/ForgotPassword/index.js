import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import cookie from 'react-cookies';
import PropTypes from 'prop-types';
import history from '../../history';
import ErrorBoundary from '../../containers/ErrorBoundary';
import ForgotPasswordForm from './ForgotPasswordForm';
import VerifyResetCodeModal from '../../components/ForgotPassword/VerifyResetCodeModal';

import {
  FORGOT_PASSWORD_REQUEST,
  VERIFY_RESET_CODE_REQUEST
} from '../../redux/actions/actionTypes';

const useStyles = makeStyles(() => ({
  h1: {
    fontSize: '36px',
    fontWeight: '900',
    textAlign: 'center',
    marginTop: '47px'
  },
  subtitle: {
    textAlign: 'center'
  },
  form: {
    textAlign: 'center',
    maxWidth: '400px',
    margin: '40px auto 80px auto'
  }
}));

function ForgotPassword(props) {
  const classes = useStyles();
  const {
    loading,
    sendContact,
    verifyResetCode,
    isContactSent,
    isResetCodeVerified
  } = props;
  const [phoneNumber, setPhoneNumber] = useState(null);

  useEffect(() => {
    if (isResetCodeVerified) {
      history.push(`/reset?token=${isResetCodeVerified}`);
    }
  }, [isResetCodeVerified]);

  try {
    const submit = values => {
      if (!/@/.test(values.contact)) {
        setPhoneNumber(values.contact);
      }
      sendContact(values);
    };

    const submitPhoneCode = values => {
      verifyResetCode({ ...values, phoneNumber });
    };

    if (cookie.load('token')) {
      history.push('/snapshot');
    }

    const contactSent = phoneNumber ? (
      <VerifyResetCodeModal onSubmit={submitPhoneCode} loading={loading} />
    ) : (
      <Typography className={classes.subtitle}>
        We have sent link to your mail, please, check it.
      </Typography>
    );

    return (
      <React.Fragment>
        <Typography className={classes.h1}>Forgot password</Typography>
        {!isContactSent ? (
          <React.Fragment>
            <Typography className={classes.subtitle}>
              We will send link to reset your password to your contact.
            </Typography>
            <div className={classes.form}>
              <ForgotPasswordForm onSubmit={submit} loading={loading} />
            </div>
          </React.Fragment>
        ) : (
          contactSent
        )}
      </React.Fragment>
    );
  } catch (e) {
    return <ErrorBoundary error={e} />;
  }
}

ForgotPassword.defaultProps = {
  loading: undefined,
  sendContact: undefined,
  verifyResetCode: undefined,
  isContactSent: undefined,
  isResetCodeVerified: undefined
};

ForgotPassword.propTypes = {
  loading: PropTypes.bool,
  sendContact: PropTypes.func,
  verifyResetCode: PropTypes.func,
  isContactSent: PropTypes.bool,
  isResetCodeVerified: PropTypes.bool
};

const mapStateToProps = state => ({
  loading: state.loading,
  isContactSent: state.isContactSent,
  isResetCodeVerified: state.isResetCodeVerified
});

const actionsStateToProps = {
  sendContact: data => ({ type: FORGOT_PASSWORD_REQUEST, data }),
  verifyResetCode: data => ({ type: VERIFY_RESET_CODE_REQUEST, data })
};

export default connect(mapStateToProps, actionsStateToProps)(ForgotPassword);
