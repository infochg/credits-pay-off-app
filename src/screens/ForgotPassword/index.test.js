import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import ForgotPassword from './index';

describe('<ForgotPassword /> before sending email', () => {
  const mockStore = configureStore();
  let wrapper;

  beforeEach(() => {
    const store = mockStore({
      isContactSent: false
    });

    wrapper = mount(
      <Provider store={store}>
        <ForgotPassword />
      </Provider>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    expect(wrapper.text()).toContain(
      'We will send link to reset your password to your contact.'
    );
  });
});

describe('<ForgotPassword /> after sending email', () => {
  const mockStore = configureStore();
  let wrapper;

  beforeEach(() => {
    const store = mockStore({
      isContactSent: true
    });

    wrapper = mount(
      <Provider store={store}>
        <ForgotPassword />
      </Provider>
    );
  });

  it('renders correctly after sending email', () => {
    expect(wrapper.text()).toContain(
      'We have sent link to your mail, please, check it.'
    );
  });
});
