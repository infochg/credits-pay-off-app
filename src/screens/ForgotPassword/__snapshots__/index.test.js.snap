// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword /> before sending email should match the snapshot 1`] = `"<p class=\\"MuiTypography-root makeStyles-h1-1 MuiTypography-body1\\">Forgot password</p><p class=\\"MuiTypography-root makeStyles-subtitle-2 MuiTypography-body1\\">We will send link to reset your password to your contact.</p><div class=\\"makeStyles-form-3\\"><form><div class=\\"MuiGrid-root MuiGrid-container\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"makeStyles-wrapper-139\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root makeStyles-label-141 MuiInputLabel-animated\\"></label><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" name=\\"contact\\" placeholder=\\"Email or Phone number\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input makeStyles-input-140\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legend-201\\" style=\\"width: 0.01px;\\"><span>â€‹</span></legend></fieldset></div></div></div></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitBtn-34 MuiButton-containedPrimary MuiButton-disableElevation Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"submit\\" disabled=\\"\\"><span class=\\"MuiButton-label\\">Send</span></button></form></div>"`;
